build/templates/session.py.mako

```diff
@@ -126,7 +126,7 @@ init_call_params = helper.get_params_snippet(init_function, helper.ParameterUsag
 
     def __setattr__(self, key, value):
         if self._is_frozen and key not in dir(self):
-            raise TypeError("%r is a frozen class" % self)
+            raise AttributeError("'Session' object has no attribute '{0}'".format(key))
```

Comment:
What error do you get if you write the wrong attribute on:

    session['0'].potato

Thing is, session['0'] is not a Session. This would also break should we ever create Session subclasses. A more robust solution would be to use type(self).

Reply:
Line 177 of test_modinst.py currently tests that case.

  >  def test_cannot_add_properties_to_device(self):
        with nimodinst.Session('') as session:
            try:
                session[0].nonexistent_property = 5
                assert False
            except AttributeError as e:
                assert str(e) == "Set attribute not supported."
            try:
                session[0].nonexistent_property
                assert False
            except AttributeError as e:
                assert str(e) == "'Device' object has no attribute 'nonexistent_property'"

Where would I do type(self)?

Reply:
Fixed